requre/helpers/files.py

```diff
@@ -91,11 +91,48 @@ def read_file_content(
         return content
 
     @classmethod
-    def _copy_logic(cls, cassette: Cassette, pathname: str, keys: list) -> None:
+    def __read_file(cls, content, pathname):
+        with BytesIO(content) as fileobj:
+            with tarfile.open(
+                mode=f"r:{cls.tar_compression}", fileobj=fileobj
+            ) as tar_store:
+                tarinfo_1st_member = tar_store.getmembers()[0]
+                if tarinfo_1st_member.isfile():
+                    with open(pathname, mode="wb") as output_file:
+                        output_file.write(
+                            tar_store.extractfile(tarinfo_1st_member).read()
+                        )
+                else:
+                    for tar_item in tar_store.getmembers():
+                        # we have to modify path of files to remove topdir
+                        if len(tar_item.name.split(os.path.sep, 1)) > 1:
+                            tar_item.name = tar_item.name.split(os.path.sep, 1)[1]
+                        else:
+                            tar_item.name = "."
+                        try:
+                            tar_store.extract(tar_item, path=pathname)
+                        except IOError:
+                            # rewrite readonly files if necessary
+                            os.remove(os.path.join(pathname, tar_item.name))
+                            tar_store.extract(tar_item, path=pathname)
```

Comment:
TBH, I still don't understand this part and feel like it can be dangerous when I see the `os.remove`...

Reply:
I understand. the reason is e.g. `.git` some files there are readonly (locks or some DBS's), and when do extract, it will raise exception and fail in case there are already files. BTW it is not changed, just code moved outside big function